src/main/java/frc/robot/commands/StorageRunCommand.java

```diff
@@ -14,17 +15,26 @@ public StorageRunCommand() {
 
     @Override
     public void execute() {
-        storage.run();
+        storage.runStorage();
+        storage.runTransition();
     }
 
     @Override
     public boolean isFinished() {
-        return false;
+        if (storage.noteExitingStorage()) {
+            seenNote = true;
+            return false;
+        } else if (!storage.noteEnteringStorage() && seenNote) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
This could be simplified as follows
```suggestion
        if (storage.noteExitingStorage()) {
            seenNote = true;
        } else if (seenNote) {
            return true;
        } 
        return false;
```
Because `if (noteExitingStorage)` then the else case already means `(!notExitingStorage)` so doing the check again just slows down the program a bit
Also, reducing the total number of returns from a method will make the code easier to debug and understand